src/main/java/software/amazon/nio/spi/s3/config/S3NioSpiConfiguration.java

```diff
@@ -75,38 +115,209 @@ public S3NioSpiConfiguration(){
      */
     public int getMaxFragmentSize(){
-        return parseIntProperty(S3_SPI_READ_MAX_FRAGMENT_SIZE_PROPERTY,
-                Integer.parseInt(S3_SPI_READ_MAX_FRAGMENT_SIZE_DEFAULT));
+        return parseIntProperty(
+            S3_SPI_READ_MAX_FRAGMENT_SIZE_PROPERTY,
+            S3_SPI_READ_MAX_FRAGMENT_SIZE_DEFAULT
+        );
     }
 
     /**
      * Get the value of the Maximum Fragment Number
      * @return the configured value or the default if not overridden
      */
     public int getMaxFragmentNumber(){
-        return parseIntProperty(S3_SPI_READ_MAX_FRAGMENT_NUMBER_PROPERTY,
-                Integer.parseInt(S3_SPI_READ_MAX_FRAGMENT_NUMBER_DEFAULT));
+        return parseIntProperty(
+            S3_SPI_READ_MAX_FRAGMENT_NUMBER_PROPERTY,
+            S3_SPI_READ_MAX_FRAGMENT_NUMBER_DEFAULT
+        );
     }
 
-    private int parseIntProperty(String propName, int defaultVal){
-        String propertyVal = properties.getProperty(propName);
-        try{
-            return Integer.parseInt(propertyVal);
-        } catch (NumberFormatException e){
-            logger.warn("the value of '{}' for '{}' is not an integer, using default value of '{}'",
-                    propertyVal, propName, defaultVal);
-            return defaultVal;
+    /**
+     * Get the value of the endpoint. Not that no endvar/sysprop is taken as
+     * default.
+     *
+     * @return the configured value or the default ("") if not overridden
+     */
+    public String getEndpoint() {
+        return getOrDefault(S3_SPI_ENDPOINT_PROPERTY, S3_SPI_ENDPOINT_DEFAULT);
+    }
+
+    /**
+     * Get the value of the endpoint protocol
+     * @return the configured value or the default if not overridden
+     */
+    public String getEndpointProtocol() {
+        String protocol = getOrDefault(S3_SPI_ENDPOINT_PROTOCOL_PROPERTY, S3_SPI_ENDPOINT_PROTOCOL_DEFAULT);
+        if ("http".equalsIgnoreCase(protocol) || "https".equalsIgnoreCase(protocol)) {
+            return protocol;
         }
+        logger.warn("the value of '{}' for '{}' is not 'http'|'https', using default value of '{}'",
+                    protocol, S3_SPI_ENDPOINT_PROTOCOL_PROPERTY, S3_SPI_ENDPOINT_PROTOCOL_DEFAULT);
+        return S3_SPI_ENDPOINT_PROTOCOL_DEFAULT;
+    }
+
+    /**
+     * Get the configured credentials
+     * @return the configured value or null if not provided
+     */
+    public AwsCredentials getCredentials() {
+        if (containsKey(AWS_ACCESS_KEY_PROPERTY)) {
+            return AwsBasicCredentials.create(
+                get(AWS_ACCESS_KEY_PROPERTY),
+                get(AWS_SECRET_ACCESS_KEY_PROPERTY)
+           );
+        }
+
+        return null;
+    }
+
+    /**
+     * Get the configured region if any
+     *
+     * @return the configured value or null if not provided
+     */
+    public String getRegion() {
+        return get(AWS_REGION_PROPERTY);
     }
 
+    // ------------------------------------------------------- protected methods
```

Comment:
remove this comment, we haven't used this convention in other places so probably shouldn't introduce it here